src/lib.rs

```diff
@@ -1,57 +1,174 @@
 //! # reedline `\|/`
-//! A readline replacement written in Rust
+//! # A readline replacement written in Rust
 //!
-//! ## Example (Simple REPL)
+//! ## Basic example
 //!
 //! ```rust,no_run
-//! use reedline::{Reedline, DefaultPrompt, Signal};
+//! // Create a default reedline object to handle user input
 //!
-//! let mut line_editor = Reedline::new();
-//! let prompt = DefaultPrompt::default();
+//! use reedline::{DefaultPrompt, Reedline, Signal};
 //!
-//! loop {
-//!     let sig = line_editor.read_line(&prompt).unwrap();
-//!     match sig {
-//!         Signal::CtrlD | Signal::CtrlC => {
-//!             line_editor.print_crlf().unwrap();
-//!             break;
-//!         }
-//!         Signal::Success(buffer) => {
-//!             // process `buffer`
-//!             println!("We processed: {}", buffer);
-//!         }
-//!         Signal::CtrlL => {
-//!             line_editor.clear_screen().unwrap();
+//! fn main() {
+//!     let mut line_editor = Reedline::new();
+//!     let prompt = DefaultPrompt::default();
+//!
+//!     loop {
+//!         let sig = line_editor.read_line(&prompt).unwrap();
+//!         match sig {
+//!             Signal::Success(buffer) => {
+//!                 println!("We processed: {}", buffer);
+//!             }
+//!             Signal::CtrlD | Signal::CtrlC => {
+//!                 line_editor.print_crlf().unwrap();
+//!                 break;
+//!             }
+//!             Signal::CtrlL => {
+//!                 line_editor.clear_screen().unwrap();
+//!             }
 //!         }
 //!     }
 //! }
+//! ```
+//! ## Integrate with custom Keybindings
+//!
+//! ```rust,no_run
+//! // Configure reedline with custom keybindings
+//!
+//! //Cargo.toml
+//! //	[dependencies]
+//! //	crossterm = "*"
+//!
+//! use {
+//!   crossterm::event::{KeyCode, KeyModifiers},
+//!   reedline::{default_emacs_keybindings, EditCommand, Reedline},
+//! };
+//!
+//! let mut keybindings = default_emacs_keybindings();
+//! keybindings.add_binding(
+//! 	KeyModifiers::ALT,
+//!   KeyCode::Char('m'),
+//!   vec![EditCommand::BackspaceWord],
+//! );
+//!
+//! let mut line_editor = Reedline::new().with_keybindings(keybindings);
+//! ```
 //!
+//! ## Integrate with custom Edit Mode
+//!
+//! ```rust,no_run
+//! // Create a reedline object with custom edit mode
+//!
+//! use reedline::{EditMode, Reedline};
+//!
+//! let mut line_editor = Reedline::new().with_edit_mode(
+//!   EditMode::ViNormal, // or EditMode::Emacs
+//! );
+//! ```
```

Comment:
As vi support is currently rudimentary we might want to push this example a bit down. (Also most shells boot into `ViInsert`)

Reply:
Done.